answerfile.py

```diff
@@ -267,7 +269,21 @@ def parseSource(self):
             if rtype == 'url':
                 address = util.URL(address)
 
-            results['sources'].append({'media': rtype, 'address': address})
+            # workaround getBoolAttribute() not allowing "None" as
+            # default, by using a getStrAttribute() call first to
+            # handle the default situation where the attribute is not
+            # specified
+            repo_gpgcheck = (None if getStrAttribute(i, ['repo-gpgcheck'], default=None) is None
+                             else getBoolAttribute(i, ['repo-gpgcheck']))
```

Comment:
Why not:
```python
getStrAttribute(i, ['repo-gpgcheck'], default=None) and getBoolAttribute(i, ['repo-gpgcheck'])
```
?

Reply:
Wait a minute. Doesn't getBoolAttribute already handle None as a special default value?

```python
def getBoolAttribute(el, attrs, default = None):
    mandatory = (default == None)
    val = getStrAttribute(el, attrs, '', mandatory).lower()
    if val == '':
        return default
    return val in ['yes', 'true']
```

Reply:
Oh, there's this `mandatory` parameter though :thinking:

Reply:
The workaround is not pretty, but I don't find either of Ronan's proposals more readable, personnally (does the second one actually work?).

Reply:
Regardless of the logic operator to use (I guess you meant `or` ?), that would cause an explicit `False` on the source (meant to override the default of `True`) to be indistinguishable from `None` (meant to use the default, especially important here when not the overridden by the global setting).

Reply:
> Oh, there's this `mandatory` parameter though thinking

That's the problem, omitting the `default` parameter means making the attribute mandatory, and this is precisely detected using `None`.

But then, it seems pretty sane that `getBoolAttribute()` can only return a bool and never `None`, so I don't think we can do any better, short of introducing an explicit tristate type, which would clearly look like overkill.